src/main/scala/amba/apb/ToTL.scala

```diff
@@ -45,23 +43,44 @@ class APBToTL()(implicit p: Parameters) extends LazyModule
       require (edgeOut.manager.minLatency >= 1)
       assert (!(in.psel && !in.penable && out.d.valid))
 
+
+      val beat = TransferSizes(beatBytes, beatBytes)
+      // The double negative here is to work around Chisel's broken implementation of widening ~x.
+      val aligned_addr =  ~(~in.paddr | (beatBytes-1).U)
+      require(beatBytes == log2Ceil(in.params.dataBits/8),
+              s"TL beatBytes(${beat_bytes}) doesn't match expected APB data width(${in.params.dataBits})")
```

Comment:
There's no need to put this requirement here. It's true by your definition of beatBytes. My comment was just that it's easier to read code when the same variable name is used consistently.

Reply:
Also, your requirement is just wrong. beatBytes == in.params.dataBits/8  ... there's no log2Ceil.

Reply:
Sure, I understood your comment. But I added the require since I wasn't sure about the code at the top... if in.params.dataBits doesn't show up in this file, it's getting checked how...?

Reply:
Your adapter did not change the width of the bus.
Therefore, edgeIn.manager.beatBytes = edgeOut.manager.beatBytes = in.params.dataBits/8 = out.params.dataBits/8

Reply:
ok. If you don't mind, I'd like to leave the require since it's not obvious to me (and potentially future code readers) otherwise.

Reply:
Your call. You should fix the compile error, though. (beat_bytes => beatBytes)

Reply:
sigh. I need to stop editing code through GitHub